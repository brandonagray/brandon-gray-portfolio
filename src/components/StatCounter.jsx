/*
  STAT_COUNTER.JS
    A counter component with a description.
    Gradually increments a number until the desired value,
    used to track stats on the About page.
*/

import React from 'react';
import styled from 'styled-components';
import CountUp from 'react-countup';
import { mediaSize } from '../data/configOptions';

const CounterContainer = styled.span`
  display: inline-grid;
  grid-template-rows: auto auto;
  grid-template-areas:
    'counter'
    'desc';
  align-items: center;

  ${mediaSize.tablet`
    max-width: 100%;
  `} ${mediaSize.phone`
    padding: 5%;
    justify-items: center;
    text-align: center;
  `};
`;

const Counter = styled.span`
  font-size: 3.5vw;
  grid-area: counter;

  ${mediaSize.tablet`
    font-size: 4em;
  `} ${mediaSize.phone`
    font-size: 2em;
  `};
`;

const Desc = styled.span`
  grid-area: desc;
  font-size: 2.5vh;
  ${mediaSize.tablet`
    font-size: 1.5em;
  `} ${mediaSize.phone`
    font-size: 1em;

  `};
`;

const StatCounter = props => (
  <CounterContainer>
    <Counter>
      <CountUp
        start={props.countStart || 0}
        end={props.countEnd || 100}
        duration={props.countDuration || 250}
      />
    </Counter>
    <Desc>{props.children}</Desc>
  </CounterContainer>
);

export default StatCounter;
