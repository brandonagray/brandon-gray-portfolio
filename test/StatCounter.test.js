import React from 'react';
import StatCounter from '../src/components/StatCounter';

describe('StatCounter component', () => {
  it('should render the stat counter', () => {
    const counter = mount(
      <StatCounter countStart={0} useEasing={true}>
        "Tests Created"
      </StatCounter>
    );

    expect(counter).toMatchSnapshot();
  });
});
