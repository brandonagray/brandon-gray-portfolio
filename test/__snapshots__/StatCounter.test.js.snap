// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatCounter component should render the stat counter 1`] = `
.c0 {
  display: inline-grid;
  grid-template-rows: auto auto;
  grid-template-areas: 'counter' 'desc';
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 {
  font-size: 3.5vw;
  grid-area: counter;
}

.c2 {
  grid-area: desc;
  font-size: 2.5vh;
}

@media (max-width:64em) {
  .c0 {
    max-width: 100%;
  }
}

@media (max-width:26.5625em) {
  .c0 {
    padding: 5%;
    justify-items: center;
    text-align: center;
  }
}

@media (max-width:64em) {
  .c1 {
    font-size: 4em;
  }
}

@media (max-width:26.5625em) {
  .c1 {
    font-size: 2em;
  }
}

@media (max-width:64em) {
  .c2 {
    font-size: 1.5em;
  }
}

@media (max-width:26.5625em) {
  .c2 {
    font-size: 1em;
  }
}

<StatCounter
  countStart={0}
  useEasing={true}
>
  <styled.span>
    <span
      className="c0"
    >
      <styled.span>
        <span
          className="c1"
        >
          <CountUp
            decimal="."
            decimals={0}
            duration={250}
            easingFn={null}
            end={100}
            formattingFn={null}
            prefix=""
            redraw={false}
            separator=""
            start={0}
            suffix=""
            useEasing={true}
          >
            <span>
              0
            </span>
          </CountUp>
        </span>
      </styled.span>
      <styled.span>
        <span
          className="c2"
        >
          "Tests Created"
        </span>
      </styled.span>
    </span>
  </styled.span>
</StatCounter>
`;
